spec/components/schemas/Subscription.yaml

```
@@ -102,3 +102,10 @@ properties:
     description: Subscription activation time
     allOf:
       - $ref: "#/components/schemas/ServerTimestamp"
+  couponIds:
+    type: array
+    description: A list of coupons to redeem on the customer and restrict by this subscription
```

```suggestion
    description: A list of coupons to redeem on the customer and restrict by this subscription
      If the customer already has redeemed the coupon, it will have no effect.  
```

Is my suggestion correct?

---

No, not really. A customer can redeem a coupon multiple times (with different restrictions). So I'm doing an unconditional redemption with the mentioned restriction.

---

```suggestion
    description: A list of coupons to redeem on the customer and restrict to this subscription.
      Read more about [coupons here](https://help.rebilly.com/invoices-and-subscriptions/coupons-discounts).
```
The restriction wasn't clear to me.